modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IORunLoop.kt

```diff
@@ -273,6 +271,10 @@ internal object IORunLoop {
         null -> {
           currentIO = IO.RaiseError(NullPointerException("Looping on null IO"))
         }
+        else -> {
+          // Since we don't capture the value of `when` kotlin doesn't enforce exhaustiveness
```

Comment:
This doesn't sound correct.

Reply:
This is actually the case for all `when` blocks that are not expressions and whose RHS produce `Unit`. In that case, Kotlin does not require exhaustiveness.

Reply:
https://kotlinlang.org/docs/reference/control-flow.html
> If when is used as an __expression__, the else branch is mandatory

Reply:
👍 

Use the `exhaustive` trick then

Reply:
Oops... left a general comment instead of a reply here.